### PR TITLE
Run layer tests in pre-commit CI validation on Azure

### PR DESCRIPTION
**Pull request by alexander-shchepetov**

### CODE REVIEW

File: .ci/azure/linux.yml [hunk 1]
```diff
@@ -191,14 +194,27 @@ jobs:
...
       env
       cd $(REPO_DIR)/inference-engine/ie_bridges/python/tests
       pytest pytest --junitxml=TEST-PythonAPI.xml
     displayName: 'Python API Tests'
     continueOnError: false
     enabled: false
 
+  - script: |
+      export MO_ROOT=$(MO_ROOT)
+      export IE_APP_PATH=$(REPO_DIR)/inference-engine/
```
Comment (vurusovs) [inline .ci/azure/linux.yml:207]: can't find is `IE_APP_PATH` used in layer tests

File: .ci/azure/linux.yml [hunk 2]
```diff
@@ -191,14 +194,27 @@ jobs:
...
+      export PYTHONPATH=$(MO_ROOT):$(LAYER_TESTS_DIR):$(PYTHONPATH)
+      sudo apt-get install python3-setuptools
+      python3 -m pip install --upgrade pip
+      python3 -m pip install -r $(LAYER_TESTS_DIR)/requirements.txt
+      python3 -m pip install -r $(MO_ROOT)/requirements_tf.txt
```
Comment (vurusovs) [inline .ci/azure/linux.yml:212]: what about requirements_onnx.txt?

File: .ci/azure/linux.yml [hunk 3]
```diff
@@ -191,14 +194,27 @@ jobs:
...
+      env
+      python3 -m pytest $(LAYER_TESTS_DIR)/tensorflow_tests/test_tf_Roll.py --ir_version=10 --junitxml=TEST-tf_Roll.xml
+    displayName: 'Roll Layer Test'
```
Comment (vurusovs) [inline .ci/azure/linux.yml:215]: TF Roll

File: .ci/azure/linux.yml [hunk 4]
```diff
@@ -33,6 +33,9 @@ jobs:
     BIN_DIR: $(REPO_DIR)/bin/intel64/$(BUILD_TYPE)
     INSTALL_DIR: $(WORK_DIR)/install_pkg
     SETUPVARS: $(INSTALL_DIR)/bin/setupvars.sh
+    PYTHONPATH: $(BIN_DIR)/lib/python_api/python3.6
```
Comment (azhogov) [inline .ci/azure/linux.yml:36]: it is updated, see https://github.com/openvinotoolkit/openvino/pull/6571

File: .ci/azure/linux.yml [hunk 5]
```diff
@@ -191,14 +194,27 @@ jobs:
       export DATA_PATH=$(MODELS_PATH)
       export MODELS_PATH=$(MODELS_PATH)
       export LD_LIBRARY_PATH=$(BIN_DIR)/lib
-      export PYTHONPATH=$(BIN_DIR)/lib/python_api/python3.6
+      export PYTHONPATH=$(PYTHONPATH)
```
Comment (azhogov) [inline .ci/azure/linux.yml:197]: should not be used

File: .ci/azure/linux.yml [hunk 6]
```diff
@@ -191,14 +194,27 @@ jobs:
...
       env
       cd $(REPO_DIR)/inference-engine/ie_bridges/python/tests
       pytest pytest --junitxml=TEST-PythonAPI.xml
     displayName: 'Python API Tests'
     continueOnError: false
     enabled: false
 
+  - script: |
+      export MO_ROOT=$(MO_ROOT)
```
Comment (azhogov) [inline .ci/azure/linux.yml:206]: should not be used

File: .ci/azure/linux.yml [hunk 7]
```diff
@@ -191,14 +194,27 @@ jobs:
...
+      export IE_APP_PATH=$(REPO_DIR)/inference-engine/
+      export PYTHONPATH=$(MO_ROOT):$(LAYER_TESTS_DIR):$(PYTHONPATH)
+      sudo apt-get install python3-setuptools
```
Comment (azhogov) [inline .ci/azure/linux.yml:209]: is it not installed?

File: .ci/azure/linux.yml [hunk 8]
```diff
@@ -191,14 +194,27 @@ jobs:
...
+      python3 -m pip install --upgrade pip
```
Comment (azhogov) [inline .ci/azure/linux.yml:210]: should be in "Install dependencies" with a comment

File: .ci/azure/linux.yml [hunk 9]
```diff
@@ -191,14 +194,27 @@ jobs:
...
+      python3 -m pip install -r $(LAYER_TESTS_DIR)/requirements.txt
+      python3 -m pip install -r $(MO_ROOT)/requirements_tf.txt
+      env
```
Comment (azhogov) [inline .ci/azure/linux.yml:213]: not needed

File: .ci/azure/linux.yml (hunk 3 above)
Comment (azhogov) [reply .ci/azure/linux.yml:215]: just Layer tests

**Review by azhogov — COMMENTED** [review]
Please update PR title

File: .ci/azure/linux.yml (hunk 6 above)
Comment (vurusovs) [reply .ci/azure/linux.yml:206]: It's a specific requirement for the tests. To remove dependency, need to update tests too

**Comment by alexander-shchepetov** [comment]
@azhogov, @vurusovs, please review again

**Comment by alexander-shchepetov** [comment]
@azhogov, @vurusovs, please review again

File: scripts/setupvars/setupvars.sh [hunk 10]
```diff
@@ -103,6 +103,11 @@ if [ -e "$INTEL_OPENVINO_DIR/deployment_tools/tools/post_training_optimization_t
     export PYTHONPATH="$INTEL_OPENVINO_DIR/deployment_tools/tools/post_training_optimization_toolkit:$PYTHONPATH"
 fi
 
+if [ -e "$INTEL_OPENVINO_DIR/deployment_tools/tests" ]; then
+    export MO_ROOT="$INTEL_OPENVINO_DIR/deployment_tools/model_optimizer"
+    export PYTHONPATH="$INTEL_OPENVINO_DIR/deployment_tools:$MO_ROOT:$PYTHONPATH"
```
Comment (vurusovs) [inline scripts/setupvars/setupvars.sh:108]: Why tests need `deployment_tools` to be available in PYTHONPATH? Perhaps need to add `deployment_tools/tests`?

File: scripts/setupvars/setupvars.sh [hunk 11]
```diff
@@ -103,6 +103,11 @@ if [ -e "$INTEL_OPENVINO_DIR/deployment_tools/tools/post_training_optimization_t
     export PYTHONPATH="$INTEL_OPENVINO_DIR/deployment_tools/tools/post_training_optimization_toolkit:$PYTHONPATH"
 fi
 
+if [ -e "$INTEL_OPENVINO_DIR/deployment_tools/tests" ]; then
+    export MO_ROOT="$INTEL_OPENVINO_DIR/deployment_tools/model_optimizer"
```
Comment (azhogov) [inline scripts/setupvars/setupvars.sh:107]: We should not add internal hard-codes into common code

File: scripts/setupvars/setupvars.sh [hunk 12]
```diff
@@ -103,6 +103,11 @@ if [ -e "$INTEL_OPENVINO_DIR/deployment_tools/tools/post_training_optimization_t
...
+    export PYTHONPATH="$INTEL_OPENVINO_DIR/deployment_tools/tests/layer_tests:$MO_ROOT:$PYTHONPATH"
```
Comment (azhogov) [inline scripts/setupvars/setupvars.sh:108]: test env should not be in the common code

File: .ci/azure/linux.yml [hunk 13]
```diff
@@ -236,6 +253,16 @@ jobs:
       . $(SETUPVARS) -pyver 3.8 && python3 -m pytest --junitxml=TEST-PythonAPI.xml
     displayName: 'Python API Tests'
     continueOnError: false
+    
+  - script: |
+      . $(SETUPVARS)
+      python3 -m pip install -r requirements.txt
+      export MO_ROOT=$(INSTALL_DIR)/deployment_tools/model_optimizer
```
Comment (azhogov) [inline .ci/azure/linux.yml:260]: tests should be updated to avoid use MO_ROOT

File: .ci/azure/linux.yml [hunk 14]
```diff
@@ -236,6 +253,16 @@ jobs:
...
+      export PYTHONPATH=$(LAYER_TESTS_DIR):$MO_ROOT:$PYTHONPATH
```
Comment (azhogov) [inline .ci/azure/linux.yml:261]: MO_ROOT is not needed

**Review by azhogov — APPROVED** [review]

**Pull request merged**